### PR TITLE
feat(enhancement): Fighter personalities

### PR DESCRIPTION
**Pull request by Quantumshark**
**Feature:** This PR implements the feature request detailed and discussed in issues #1147 and #3098

## Feature Details
Allows you to set different personalities for a fleet's fighters, seperate from the personalities used by the rest of the fleet. Also refactors fighter names to be a child of `fighters`, rather than having `fighters` mean fighter names. The old system is still supported, so plugins won't break, but if you mix the two then only the phrase given after `fighters` is read (as before), while any child nodes it may have are ignored.

## UI Screenshots
N/A

## Usage Examples
```
fleet "Large Republic"
	government "Republic"
	names "republic capital"
	fighters
		names "republic fighter"
		personality
			heroic target
	personality
		heroic opportunistic
```

## Testing Done
Loaded up the game and confirmed that it didn't crash and hadn't broken support for the existing names.
Switched the Large Republic fleet to the new syntax and gave the fighters `target`, and confirmed that they correctly used the "republic fighter" names and the `target` personality, while the other ships in the fleet still used "republic capital" and lacked "target".

### Automated Tests Added
Added a check to the data files that spots use of the old fighter name syntax.

## Performance Impact
None observed, and I don't think any significant impact is likely.

### CODE REVIEW

**Comment by Quantumshark** [comment]
I should probably have updated the data files before opening this. Oops.

**Comment by Zitchas** [comment]
This looks useful. I can think of a variety of places where it'd be good to have fighters and fleet have different personalities.

File: source/Fleet.cpp [hunk 1]
```diff
@@ -88,7 +88,24 @@ void Fleet::Load(const DataNode &node)
...
+		{
+			for(const DataNode &grand : child)
+			{
+				const string &fighterKey = grand.Token(0);
+				if(fighterKey == "names" && grand.Size() >= 2)
+					fighterNames = GameData::Phrases().Get(grand.Token(1));
+				else if(fighterKey == "personality")
+					fighterPersonality.Load(grand);
+				else
+					grand.PrintTrace("Skipping unrecognized attribute under fighters:");
```
Comment (RisingLeaf) [inline source/Fleet.cpp:106]: do we normally add an "under [parent node]"? I thought this automatically gets added by the print trace

File: source/Fleet.cpp [hunk 2]
```diff
@@ -469,11 +486,15 @@ vector<shared_ptr<Ship>> Fleet::Instantiate(const vector<const Ship *> &ships) c
 		// Copy the model instance into a new instance.
 		auto ship = make_shared<Ship>(*model);
 
-		const Phrase *phrase = ((ship->CanBeCarried() && fighterNames) ? fighterNames : names);
+		bool isCarried = ship->CanBeCarried();
```
Comment (RisingLeaf) [inline source/Fleet.cpp:489]: misleading name, why not say canBeCarried?

**Review by RisingLeaf — COMMENTED** [review]
I generally despise doing the content changes alongside the code changes, but that's probably just me.

The code side looks mostly fine, two minor things.

**Comment by Hurleveur** [comment]
The content changes were done to adapt to the code changes and the depreciation of the old way to do it @RisingLeaf

**Comment by RisingLeaf** [comment]
oh, my bad

**Review by RisingLeaf — APPROVED** [review]
I think this is good to go, all I need is a comment on the print trace thing

File: source/Fleet.cpp (hunk 1 above)
Comment (Quantumshark) [reply source/Fleet.cpp:106]: You are correct, the print trace makes it clear what the parent node is without needing to specify here.

File: data/korath/korath.txt [hunk 3]
```diff
@@ -692,7 +708,8 @@ fleet "Large Kor Sestor"
 fleet "Small Kor Mereti"
 	government "Kor Mereti"
 	names "kor mereti"
-	fighters "kor mereti"
+	fighters
+		names "kor mereti"
```
Comment (warp-core) [inline data/korath/korath.txt:712]: What fighters?????????

**Review by warp-core — APPROVED** [review]

**Comment by quyykk** [comment]
Why not have something like this:

```
fighters "republic fighter"
		personality
			heroic target
```

Avoids having to deprecate the old syntax.

**Comment by Hurleveur** [comment]
That seems less resilient?
Like say we want to add more attributes to it

Also its a bit weird to just have the name of the fighters under "fighters" directly to begin with, so good riddance of a bad syntax?

**Pull request merged**